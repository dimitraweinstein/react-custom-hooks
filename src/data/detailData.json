{
  "_id": "5da237699734fdcb7bef8f59",
  "name": "Grandpa's grandfather",
  "image": "https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
}