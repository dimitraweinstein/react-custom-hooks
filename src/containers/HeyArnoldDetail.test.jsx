import React from 'react';
import { render, screen } from '@testing-library/react';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import HeyArnoldDetails from './HeyArnoldDetailList';
import { MemoryRouter } from 'react-router-dom';
import detailData from '../data/detailData.json';

const server = setupServer(
  rest.get(
    'https://hey-arnold-api.herokuapp.com/api/v1/characters/:id',
    (req, res, ctx) => {
      return res(ctx.json(detailData));
    })
);


describe('Hey Arnold Details List Container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());



  it('renders a single character on the page', async () => {
    const component = render(
      <MemoryRouter initialEntries={['/5da237699734fdcb7bef8f59']}>
        <HeyArnoldDetails />
      </MemoryRouter>
    );

    screen.getByText('Loading...');

    await screen.findByRole('character', { name: 'character' });
    expect(component).toMatchSnapshot();

    // const a = await screen.findByRole('link');
    // expect(a).toBeInTheDocument();

  });
});


