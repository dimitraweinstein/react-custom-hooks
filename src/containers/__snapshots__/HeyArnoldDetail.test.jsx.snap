// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hey Arnold Details List Container renders a single character on the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <figure
        aria-label="character"
        role="character"
      >
        <img
          alt="Grandpa's grandfather"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
        />
        <figcaption>
          Grandpa's grandfather
        </figcaption>
      </figure>
    </div>
  </body>,
  "container": <div>
    <figure
      aria-label="character"
      role="character"
    >
      <img
        alt="Grandpa's grandfather"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
      />
      <figcaption>
        Grandpa's grandfather
      </figcaption>
    </figure>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
